### DISCUSSION TITLE
ISSUE- Compile and Test MOOSE

### DISCUSSION THREAD

**Discussion by hyeom-wisc** [idaholab/moose#16279] [Q&A Getting Started] [answered]
Hello, 
I am a new user for MOOSE frame code and have followed the instructions to install the code. After cloning MOOSE, I tried to compile and test it but it kept saying Command 'make' not found. I did the instruction shown in the Troubeshooting for "Build Isses" but it still doesn't work. I would appreciate if you guide me to move forward.

**(moose) hyeom@ERB914-7920:~/projects/moose/test$ make -j 4

Command 'make' not found, but can be installed with:

sudo apt install make        # version 4.2.1-1.2, or
sudo apt install make-guile  # version 4.2.1-1.2**

**Comment by GiudGiud** [comment]
Hello

Do you have 'make' installed? What does 'which make' return?
If not you should follow the prompt, as it is required to install MOOSE

Best,
Guillaume

**Reply by hyeom-wisc** [reply]
Thank you. The issue was resolved after install the build-essential package. Sung

**Comment by milljm** [comment, marked as answer]
You probably need to install the build-essential package for Ubuntu:

```bash
sudo apt install build-essential
```

While we're on the subject of Ubuntu and installation packages, you should also install libquadmath:

```bash
sudo apt install libquadmath0
```

Once the above has completed, then try to build MOOSE.

**Reply by hyeom-wisc** [reply]
Dear milljm,
Thank you so much. The test MOOSE was successful after following your instruction. Thanks. Sung
